### PR TITLE
NP-637 conan v2 curaengine 5 9

### PR DESCRIPTION
**Pull request by jellespijker**
Publish a NPM package from a conan recipe.

> [!IMPORTANT]
> This branch is based on the 5.9 branch and does a very naive port of the workflows from `main` this is needed because we need a stable 5.9 version of the engine but the workflows and conan v2 as well.
>
> **Do not kill this branch after merge**

Renamed the `conan-package.yml` workflow to `package.yml` since it is now also used for NPM packages

Added a `conf` to the recipe, which is used by the conan npm generator to generate a `package.json`. NPM packages are created on pushes to `main`, `NP-*` and `5.10` release branch and they will end up here: https://github.com/Ultimaker/CuraEngine/pkgs/npm/curaenginejs

_Yes, the hardcoded `5.10` hurts_. 
_Yes, the replacement of the `+` to `-` in the version of the `package.json` hurts_, but NPM kept "sanitizing" the build-meta data away and the version needs to be unique.

See also:
- https://github.com/Ultimaker/cura-workflows/pull/32 Which adds a npm-package workflow, and allows for conan config test branches
- https://github.com/Ultimaker/conan-config/pull/20 Which adds a npm generator

# Checklist:

- [X] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta)
- [X] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/contributing.md)
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have uploaded any files required to test this change

### CODE REVIEW

**Review by wawanbreton — CHANGES_REQUESTED** [review]
Small remarks, but overall looks lood

File: .github/workflows/gcodeanalyzer.yml [hunk 1]
```diff
@@ -73,95 +56,31 @@ jobs:
...
         run: |
+          source build/Release/generators/conanbuild.sh
           cmake --preset release
           cmake --build --preset release
-        working-directory: CuraEngine
 
       - name: Collect STL-files, run CuraEngine, output GCode-files
         run: |
-          for file in `ls ../NightlyTestModels/*.stl`;
+          for file in `ls NightlyTestModels/*.stl`;
```
[the remark this answers is not in the document]
Comment (jellespijker) [reply .github/workflows/gcodeanalyzer.yml:78]: I just did a poor-man cherry pick from `main`, but looking at the gcodeanalyzer workflow as it is here I would say yes

**Comment by jellespijker** [comment]
Closing this PR in favor of the one based on `main` https://github.com/Ultimaker/CuraEngine/pull/2184

>[!CAUTION]
> DO NOT KILL THE BRANCH, since we need a 5.9 branch with the Conan v2 workflows

File: .github/workflows/gcodeanalyzer.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wawanbreton) [reply .github/workflows/gcodeanalyzer.yml:79]: I thought that was one of your changes, but the script is probably still broken at the moment, so you can ignore this remark.

**Pull request closed without merging**